### PR TITLE
[TikTokUser] Use mobile API + [TikTok(Base)] Cleanup

### PR DESCRIPTION
**Pull request by MinePlayersPE**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (Again, special thanks to @llacb47)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

Use mobile API instead of the (currently broken) web API to fetch user posts.
Also adds TikTokBaseIE/`_call_api` to cleanup API use and `_parse_aweme_video` to generalize video extraction

### CODE REVIEW

File: yt_dlp/extractor/tiktok.py [hunk 1]
```diff
@@ -323,38 +336,32 @@ def _real_extract(self, url):
         raise ExtractorError('Video not available', video_id=video_id)
 
 
-class TikTokUserIE(InfoExtractor):
+class TikTokUserIE(TikTokBaseIE):
     IE_NAME = 'tiktok:user'
-    _VALID_URL = r'https?://(?:www\.)?tiktok\.com/@(?P<id>[\w\._]+)/?(?:$|[#?])'
+    _VALID_URL = r'https?://(?:www\.)?tiktok\.com/@(?P<id>[\w\.-]+)/?(?:$|[#?])'
```
Comment (Ashish0804) [inline yt_dlp/extractor/tiktok.py:341]: are u sure about this change?
I think when i added this, i found some accounts using `_` in name

Comment (MinePlayersPE) [reply yt_dlp/extractor/tiktok.py:341]: `\w` catches `_` as well

**Comment by MattWing1510** [comment]
Heya 👋. Wanna let you know that this procudes the following error: "Unable to extract user ID".

**Comment by Ashish0804** [comment]
> Heya 👋. Wanna let you know that this procudes the following error: "Unable to extract user ID".

show log
and command u used

**Comment by MattWing1510** [comment]
> > Heya 👋. Wanna let you know that this procudes the following error: "Unable to extract user ID".
> 
> show log
> and command u used

Not much to see. YT and others are working - or I'm just too inexperienced with this.

`PS C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp> py.exe .\__main__.py -v https://www.tiktok.com/@_rl9?lang=de-DE
[debug] Command-line config: ['-v', 'https://www.tiktok.com/@_rl9?lang=de-DE']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.09.02 (source)
[debug] Plugin Extractors: ['SamplePlugin']
[debug] Python version 3.9.7 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] exe versions: none
[debug] Optional libraries: pycryptodome, sqlite
[debug] Proxy map: {}
[debug] [tiktok:user] Extracting URL: https://www.tiktok.com/@_rl9?lang=de-DE
[tiktok:user] _rl9: Downloading webpage
ERROR: [tiktok:user] _rl9: Unable to extract user ID; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 584, in extract
    ie_result = self._real_extract(url)
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\tiktok.py", line 412, in _real_extract
    own_id = self._html_search_regex(r'snssdk1180://user/profile/(\d+)', webpage, 'user ID')
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 1156, in _html_search_regex
    res = self._search_regex(pattern, string, name, default, fatal, flags, group)
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 1147, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
yt_dlp.utils.RegexNotFoundError: Unable to extract user ID; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 584, in extract
    ie_result = self._real_extract(url)
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\tiktok.py", line 412, in _real_extract
    own_id = self._html_search_regex(r'snssdk1180://user/profile/(\d+)', webpage, 'user ID')
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 1156, in _html_search_regex
    res = self._search_regex(pattern, string, name, default, fatal, flags, group)
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 1147, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
yt_dlp.utils.RegexNotFoundError: Unable to extract user ID; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\YoutubeDL.py", line 1218, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\YoutubeDL.py", line 1243, in __extract_info
    ie_result = ie.extract(url)
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 600, in extract
    raise ExtractorError(
yt_dlp.utils.ExtractorError: [tiktok:user] _rl9: Unable to extract user ID; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
`

**Comment by Ashish0804** [comment]
idk wtf is happening 
`https://www.tiktok.com/@_rl9` trying this in browser results in downloading an random file.
Can you open this url in browser?

**Comment by nyuszika7h** [comment]
I get this, after solving the captcha the user profile loads, no random file download.

image

**Comment by MattWing1510** [comment]
And this is my Browser few ....

`https://i.imgur.com/NtKCrRs.png`

**Comment by Ashish0804** [comment]
weird now i can't see any profiles
I think TT is broken/down in asia
I am using a usa vpn and now it works in the browser

Maybe some of the parameters used are asia only

**Comment by Ashish0804** [comment]
@MattWing1510 It works for me now, can u try again
Here's log
```
[debug] Command-line config: ['-F', '-v', '--list-subs', 'https://www.tiktok.com/@_rl9?lang=de-DE']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.09.02 (source)
[debug] Plugin Extractors: ['SamplePlugin']
[debug] Git HEAD: f827cfcaf
[debug] Python version 3.8.8 (CPython 64bit) - Windows-10-10.0.19041-SP0
[debug] exe versions: ffmpeg 3.0.1, ffprobe 3.0.1
[debug] Optional libraries: keyring, mutagen, pycryptodome, sqlite, websockets
[debug] Proxy map: {}
[debug] [tiktok:user] Extracting URL: https://www.tiktok.com/@_rl9?lang=de-DE
[tiktok:user] _rl9: Downloading webpage
[download] Downloading playlist: _rl9
[tiktok:user] _rl9: Downloading user video list page 1
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[tiktok:user] _rl9: Downloading user video list page 2
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[debug] Sort order given by extractor: quality, source, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, source, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, asr, proto, vext, aext, hasaud, id
[tiktok:user] playlist _rl9: Downloading 24 videos
[download] Downloading video 1 of 24
```

**Comment by MattWing1510** [comment]
> [debug] Git HEAD: f827cfcaf

I copied your exact params ... same issue on my side.

`PS C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp> py.exe .\__main__.py -F -v --list-subs https://www.tiktok.com/@_rl9?lang=de-DE
[debug] Command-line config: ['-F', '-v', '--list-subs', 'https://www.tiktok.com/@_rl9?lang=de-DE']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.09.02 (source)
[debug] Plugin Extractors: ['SamplePlugin']
[debug] Python version 3.9.7 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] exe versions: none
[debug] Optional libraries: pycryptodome, sqlite
[debug] Proxy map: {}
[debug] [tiktok:user] Extracting URL: https://www.tiktok.com/@_rl9?lang=de-DE
[tiktok:user] _rl9: Downloading webpage
ERROR: [tiktok:user] _rl9: Unable to extract user ID; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 584, in extract
    ie_result = self._real_extract(url)
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\tiktok.py", line 412, in _real_extract
    own_id = self._html_search_regex(r'snssdk1180://user/profile/(\d+)', webpage, 'user ID')
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 1156, in _html_search_regex
    res = self._search_regex(pattern, string, name, default, fatal, flags, group)
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 1147, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
yt_dlp.utils.RegexNotFoundError: Unable to extract user ID; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 584, in extract
    ie_result = self._real_extract(url)
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\tiktok.py", line 412, in _real_extract
    own_id = self._html_search_regex(r'snssdk1180://user/profile/(\d+)', webpage, 'user ID')
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 1156, in _html_search_regex
    res = self._search_regex(pattern, string, name, default, fatal, flags, group)
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 1147, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
yt_dlp.utils.RegexNotFoundError: Unable to extract user ID; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\YoutubeDL.py", line 1218, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\YoutubeDL.py", line 1243, in __extract_info
    ie_result = ie.extract(url)
  File "C:\Users\USER\Downloads\yt-dlp-owome-v2\yt-dlp-owome-v2\yt_dlp\extractor\common.py", line 600, in extract
    raise ExtractorError(
yt_dlp.utils.ExtractorError: [tiktok:user] _rl9: Unable to extract user ID; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
`

**Comment by Ashish0804** [comment]
Yeah it means you are getting one of those random 0 byte errors
in our internal testing aswell, tiktok user scraping is mildly broken at best.
If this makes it to release anytime soon, users will have to deal with such weird errors by keep retrying

My assumption is they are purposefully breaking site to stop scraping scripts.
I recommend you keep retrying and maybe see if they give you any captchas in site to "allow" your ip without any cookies.
(make sure to clear your cookies before checking for captchas)

**Comment by MinePlayersPE** [comment]
@MattWing1510 Sorry for pinging but can you try again

File: yt_dlp/extractor/tiktok.py [hunk 2]
```diff
@@ -199,6 +101,7 @@ def extract_addr(addr, add_meta={}):
                 'filesize': int_or_none(addr.get('data_size')),
                 'ext': 'mp4',
                 'acodec': 'aac',
+                'source_preference': -2 if 'aweme/v1' in url else -1,  # Downloads from API might get blocked
```
Comment (MinePlayersPE) [inline yt_dlp/extractor/tiktok.py:104]: Doesn't seem to actually deprioritize it :/ Any ideas?

**Comment by MattWing1510** [comment]
> @MattWing1510 Sorry for pinging but can you try again

Working like a charm right now :>

File: yt_dlp/extractor/tiktok.py (hunk 2 above)
Comment (MinePlayersPE) [reply yt_dlp/extractor/tiktok.py:104]: NVMD It does, it's just not reflected on the format selection list

File: yt_dlp/extractor/tiktok.py [hunk 3]
```diff
@@ -17,123 +18,23 @@
...
+        real_query = {
+            **query,
             'version_name': self._APP_VERSION,
             'version_code': self._MANIFEST_APP_VERSION,
             'build_number': self._APP_VERSION,
             'manifest_version_code': self._MANIFEST_APP_VERSION,
             'update_version_code': self._MANIFEST_APP_VERSION,
             'openudid': ''.join(random.choice('0123456789abcdef') for i in range(16)),
+            # Some endpoints don't like randomized device_id, while some require it.
+            # 'device_id': ''.join(random.choice(string.digits) for i in range(19)),
```
Comment (pukkandan) [inline yt_dlp/extractor/tiktok.py:37]: remove

File: yt_dlp/extractor/tiktok.py [hunk 4]
```diff
@@ -220,7 +123,7 @@ def extract_addr(addr, add_meta={}):
                 'vcodec': 'h264',
                 'width': video_info.get('width'),
                 'height': video_info.get('height'),
-                'source_preference': -2 if video_info.get('has_watermark') else -1,
+                'preference': -2 if video_info.get('has_watermark') else -1,
```
Comment (pukkandan) [inline yt_dlp/extractor/tiktok.py:126]:
```
                    * preference Order number of this format. If this field is
                                 present and not None, the formats get sorted
                                 by this field, regardless of all other values.
```
Are you sure this is what you want? The priority of `preference` cannot be adjusted using `_sort_formats`. I believe this should in fact be `source_preference` here

File: yt_dlp/extractor/tiktok.py [hunk 5]
```diff
@@ -296,6 +199,117 @@ def extract_addr(addr, add_meta={}):
...
+            'uploader_url': f'https://www.tiktok.com/@{user_id}',
+            'track': str_or_none(music_info.get('title')),
+            'album': str_or_none(music_info.get('album')) or None,
+            'artist': str_or_none(music_info.get('authorName')),
+            'thumbnails': thumbnails,
+            'description': str_or_none(video_info.get('desc')),
+            'webpage_url': self._og_search_url(webpage),
+            'http_headers': {
+                'Referer': url,
+                'Cookie': 'tt_webid=%s; tt_webid_v2=%s' % (tracker, tracker),
```
Comment (pukkandan) [inline yt_dlp/extractor/tiktok.py:304]: are you sure this is needed? The cookiejar is automatically used for download as well

File: yt_dlp/extractor/tiktok.py [hunk 6]
```diff
@@ -323,38 +337,32 @@ def _real_extract(self, url):
...
-        'skip': 'Cookies (not necessarily logged in) are needed.'
     }]
 
-    def _entries(self, url, user_id):
-        webpage = self._download_webpage(url, user_id)
-        own_id = self._search_regex(r'\"id\":\"(?P<userid>\d+)', webpage, user_id, default=None)
-        if not own_id:
-            raise ExtractorError('Cookies (not necessarily logged in) are needed.', expected=True)
-        secuid = self._search_regex(r'\"secUid\":\"(?P<secUid>[^\"]+)', webpage, user_id)
+    def _entries(self, webpage, user_id, username):  # TODO: Fix?
```
Comment (pukkandan) [inline yt_dlp/extractor/tiktok.py:357]: what's with the `TODO`?

File: yt_dlp/extractor/tiktok.py [hunk 7]
```diff
@@ -363,6 +371,44 @@ def _entries(self, url, user_id):
                 break
             cursor = data_json['cursor']
 
+    def _entries_api(self, webpage, user_id, username):
+        query = {
+            'user_id': user_id,
+            'count': 21,
+            # 'cursor': 0,
```
Comment (pukkandan) [inline yt_dlp/extractor/tiktok.py:378]: dont leave commented code

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/tiktok.py (hunk 5 above)
Comment (MinePlayersPE) [reply yt_dlp/extractor/tiktok.py:304]: ¯\\\_(ツ)_/¯ Leftover from [youtube-dlc](https://github.com/blackjack4494/yt-dlc/commit/4f5a0ad82214456c3d4832bbf4e15e416e1a16cb) and I haven't messed with webpage-based extraction besides extracting additional music info

File: yt_dlp/extractor/tiktok.py (hunk 6 above)
Comment (MinePlayersPE) [reply yt_dlp/extractor/tiktok.py:357]: https://github.com/yt-dlp/yt-dlp/issues/859

Comment (pukkandan) [reply yt_dlp/extractor/tiktok.py:355]: is this PR not addressing that issue?

Comment (MinePlayersPE) [reply yt_dlp/extractor/tiktok.py]: (resolved on discord https://github.com/yt-dlp/yt-dlp/pull/925/commits/143ca2c32118e040d90deb3b972ca4ad32ff8b7e)

**Pull request merged**